src/main/java/bio/terra/service/snapshot/SnapshotTableDao.java

```diff
@@ -27,20 +29,36 @@ public class SnapshotTableDao {
   private static final Logger logger = LoggerFactory.getLogger(SnapshotTableDao.class);
 
   private static final String sqlInsertTable =
-      "INSERT INTO snapshot_table "
-          + "(name, parent_id, primary_key) "
-          + "VALUES (:name, :parent_id, :primary_key)";
+      """
+  INSERT INTO snapshot_table
+  (name, parent_id, primary_key)
+  VALUES (:name, :parent_id, :primary_key)
+  """;
+
   private static final String sqlInsertColumn =
-      "INSERT INTO snapshot_column "
-          + "(table_id, name, type, array_of, required, ordinal) "
-          + "VALUES (:table_id, :name, :type, :array_of, :required, :ordinal)";
+      """
+  INSERT INTO snapshot_column
+  (table_id, name, type, array_of, required, ordinal)
+  VALUES (:table_id, :name, :type, :array_of, :required, :ordinal)
+  """;
+
   private static final String sqlSelectTable =
-      "SELECT id, name, row_count, primary_key FROM snapshot_table WHERE parent_id = :parent_id";
+      """
+  SELECT t.id table_id, t.name table_name, t.row_count table_row_count, t.primary_key table_primary_key,
+         c.id column_id, c.name column_name, c.type column_type, c.array_of column_array_of, c.required column_required
+  FROM snapshot_table t
+     INNER JOIN snapshot_column c ON t.id = c.table_id
+  WHERE parent_id = :snapshot_id
+  ORDER BY t.ctid, c.ordinal
+  """;
+
   private static final String sqlSelectColumn =
-      "SELECT id, name, type, array_of, required "
-          + "FROM snapshot_column "
-          + "WHERE table_id = :table_id "
-          + "ORDER BY ordinal";
+      """
+  SELECT id column_id, name column_name, type column_type, array_of column_array_of, required column_required
```

Comment:
Let's say we add a column to any of these Postgres table definitions, like `snapshot_column` table (but this comment applies overall).

Right now, we'd have to make SQL changes in 3 places: `sqlInsertColumn`, `sqlSelectTable`, and `sqlSelectColumn`.  I could see it being easy to miss a modification.

Is there a way we can define Postgres SQL column definitions for reuse?

Reply:
I was torn around that.  I actually prefer this approach since it makes the SQL easier to read.  I don't love the pattern of building sql from concatenation.  I do think that the unit test is really meant to catch that type of mistake and I personally would take readability over the maintenance in this case the changes are so incredibly rare

Reply:
Hm, okay, fair enough… you're right that such changes are rare!